templates/hooks/pod_packs-volumes.yaml

```diff
@@ -0,0 +1,97 @@
+{{- if $.Values.st2.packs.volumes.enabled }}
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {{ $.Release.Name }}-st2canary-packs-volumes-validation
+  labels: {{- include "stackstorm-ha.labels" (list $ "st2canary") | nindent 4 }}
+  annotations:
+    "helm.sh/hook": pre-install,pre-upgrade,pre-rollback
+    "helm.sh/hook-weight": -5 # fairly high priority
+    "helm.sh/hook-delete-policy": hook-succeeded
+    {{- if $.Values.st2canary.annotations }}
+      {{- toYaml $.Values.st2canary.annotations | nindent 4 }}
+    {{- end }}
+spec:
+  imagePullSecrets:
+    {{- if $.Values.image.pullSecret }}
+    - name: {{ $.Values.image.pullSecret }}
+    {{- end }}
+  initContainers: []
+  containers:
+    - name: st2canary-packs-volumes-validation
+      image: '{{ template "stackstorm-ha.imageRepository" $ }}/st2actionrunner:{{ tpl $.Values.image.tag $ }}'
+      #image: busybox:1.28
+      imagePullPolicy: {{ $.Values.image.pullPolicy }}
+      {{- with $.Values.securityContext }}
+      securityContext: {{- toYaml . | nindent 8 }}
+      {{- end }}
+      # TODO: maybe use kubectl to assert the volumes have RWX mode
+      # If volume is a persistentVolumeClaim, then:
+      #   the PVC must only have ReadWriteMany in spec.accessModes
+      # If volume is something else, then validating through metadata is iffy.
+      #   azureFile, cephfs, csi, glusterfs, nfs, pvc, quobyte, need at least:
+      #     readOnly: false
+      #   ephemeral volumes could also work, ... but that config is even deeper.
+      command:
+        - 'sh'
+        - '-ec'
+        - |
+          echo Testing write permissions for packs volumes.
+          echo If this passes, the pod will automatically be deleted.
+          echo If this fails, inspect the pod for errors in kubernetes,
+          echo and then delete this st2canary pod manually.
+          echo
+          echo Testing write permissions on packs volume...
+          touch /opt/stackstorm/packs/.write-test
+          rm /opt/stackstorm/packs/.write-test
+          echo
+          echo Testing write permissions on virtualenvs volume...
+          touch /opt/stackstorm/virtualenvs/.write-test
+          rm /opt/stackstorm/virtualenvs/.write-test
+          echo
+          {{- if $.Values.st2.packs.volumes.configs }}
+          echo Testing write permissions on configs volume...
+          touch /opt/stackstorm/configs/.write-test
+          rm /opt/stackstorm/configs/.write-test
```

Comment:
The messaging here provides some info about what it tests, but it doesn't provide a message what to do if something fails.

As a user who has no idea about StackStorm I see the error that `/opt/stackstorm/virtualenvs/.write-test` failed, but it's hard to guess what it means next and what direction to take next.

How about adding a very loud error message saying that persistent volumes don't satisfy the write requirements and also printing a link to the docs https://github.com/StackStorm/stackstorm-k8s#method-2-shared-volumes.

Reply:
OK. I updated the script to hopefully be both clearer and more helpful.